### ISSUE TITLE
Specify stricter database version

### ISSUE THREAD

**Issue by martinheidegger** [pretalx/pretalx-docker#21]
Currently the mysql database version is `5` which when resolved to `5.0` is certainly too low _(it resolves to 5.7.30 at the time of writing)_. It would probably better to be stricter with the database version (maybe `5.7`? or even with a patch version?) as it would reduce the amount possible bugs.

**Comment by MikkCZ** [comment]
There was actually an opposite change made since this ticket was reported - https://github.com/pretalx/pretalx-docker/commit/70ba5424bc043128124f5d637f4b0952f4155a3e set the database container to be used to `mysql:latest`.

**Comment by rixx** [comment]
I have no opinion on this, and am happy to follow whatever people interested in using this dockerfile want – though it would seem to me that changing databases at this point would be a major step for people using this repo, and migrating from one to the other is nontrivial.

**Comment by almereyda** [comment]
I agree that following the upstream convention seems appropriate for a downstream project. If people need a derivation, they can always go down that path in our magnificient FLOSS land.

Did I understand this correctly: Changing the database (in the main `compose.yaml` manifest) for people using this repo being a major step implies that people update their local environments usually with `git pull` and `docker compose up -d`?

While the migration is not trivial, we can certainly provide different examples through well-documented `compose.{postgres,mariadb,mysql}.yaml` manifests.

For me this discussion correlates strongly with the observations and argument in https://github.com/pretalx/pretalx-docker/issues/62#issuecomment-2080227898 : with adopting more conventional patterns of handling a dockerized Django application, we can reduce the ambiguity for end users of this repository.

The migration path from MySQL to PostgreSQL could be transparent, when this initiative comes to terms:

> ## The Future of MySQL is Postgres
> *Simplifying Migrations by Reimagining MySQL as a PostgreSQL Extension*
>
> [Jonah Harris, Postgres Conference, 2024-04-19](https://postgresconf.org/conferences/2024/program/proposals/the-future-of-mysql-is-postgres)

**Comment by rixx** [comment]
Update: pretalx now requires postgreSQL or SQLite.

**Comment by martinheidegger** [comment]
The legacy file should probably reflect that:

https://github.com/pretalx/pretalx-docker/blob/8cab436aa1664bcd2d4543fcd56309979df0a21c/docker-compose.yml#L25 

But also considering the rework → https://github.com/pretalx/pretalx-docker/pull/64 (cc. @almereyda) it may be good to be a bit more specific with the working postgres version _(15 is quite broad still, though if postgres follows semver it should be fine)_

**Comment by rixx** [comment]
PostgreSQL does not follow SemVer.

Just as a heads-up, my current impression is that this repo is in a permanently uncared-for / neigh-unusable state. If this doesn’t change by the end of the year, or at least looks like it’s likely to change by then, I’ll probably archive it. The whole “community maintained” is not really working out and leads to confusion and frustration when people come here under the impression they’ll be able to "just" run pretalx with docker, and then things don’t work or break.

**Comment by saschafoerster** [comment]
Switched the docker-compose to PostgreSQL.
